src/library_pthread.js

```diff
@@ -1042,7 +1042,11 @@ var LibraryPThread = {
 
   _emscripten_notify_proxying_queue: function(targetThreadId, currThreadId, mainThreadId, queue) {
     if (targetThreadId == currThreadId) {
-      setTimeout(function() { _emscripten_proxy_execute_queue(queue); });
+      setTimeout(() => {
+        if (_pthread_self()) {
```

Comment:
This looks like the only place in this file that we depend on `pthread_self()` returning zero to indicate something special. Might be worth a comment, as this depends on special internals of how we implement pthreads I believe.